app/src/full/java/io/homeassistant/companion/android/settings/views/SettingsWearFavoritesView.kt

```diff
@@ -0,0 +1,140 @@
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.material.Checkbox
+import androidx.compose.material.Icon
+import androidx.compose.material.IconButton
+import androidx.compose.material.Scaffold
+import androidx.compose.material.Text
+import androidx.compose.material.TopAppBar
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.HelpOutline
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import io.homeassistant.companion.android.R
+import io.homeassistant.companion.android.common.data.integration.Entity
+import io.homeassistant.companion.android.util.previewEntityList
+import io.homeassistant.companion.android.util.previewFavoritesList
+
+const val WEAR_DOCS_LINK = "https://companion.home-assistant.io/docs/wear-os/wear-os"
+val supportedDomains = listOf(
+    "input_boolean", "light", "switch", "script", "scene"
+)
+
+@Composable
+fun LoadWearFavoritesSettings(
+    entities: Map<String, Entity<*>>,
+    favoritesList: List<String>,
+    onEntitySelected: (Boolean, String, Activity) -> Unit,
+    isFavoriteSelected: (Int) -> Boolean,
+    activity: Activity
+) {
+    val context = LocalContext.current
+
+    val validEntities = entities.filter { it.key.split(".")[0] in supportedDomains }.values.sortedBy { it.entityId }.toList()
+    Scaffold(
+        topBar = {
+            TopAppBar(
+                title = { Text(stringResource(R.string.wear_favorite_entities)) },
+                actions = {
+                    IconButton(onClick = {
+                        val intent = Intent(Intent.ACTION_VIEW, Uri.parse(WEAR_DOCS_LINK))
+                        context.startActivity(intent)
+                    }) {
+                        Icon(
+                            Icons.Filled.HelpOutline,
+                            contentDescription = stringResource(id = R.string.help)
+                        )
+                    }
+                }
+            )
+        }
+    ) {
+        LazyColumn(
+            verticalArrangement = Arrangement.Center,
+            modifier = Modifier.padding(top = 10.dp, start = 20.dp, end = 20.dp)
+        ) {
+            item {
+                Text(
+                    text = stringResource(R.string.wear_set_favorites),
+                    fontWeight = FontWeight.Bold
+                )
+            }
+            items(favoritesList.size) { index ->
+                Row(
+                    modifier = Modifier
+                        .padding(15.dp)
+                        .clickable {
+                            onEntitySelected(
+                                isFavoriteSelected(index),

```

Comment:
So to add to the discussion in https://github.com/home-assistant/android/pull/1908/files/5f58c8f00b5e07da265da5239ac8fdeb0abd60bf#r750641450

This line is where that function is called, as well as in line 88.

You only reach this line for items in `favoritesList`, since that's what you loop over in line 75. So this `Row` belongs to an entity in the `favoritesList`, which in turn is `settingsWearViewModel.favoriteEntityIds.toList()`. Now when a user clicks the `Row`, a check is done on whether the clicked item is a favorite. This check takes the item from `settingsWearViewModel.favoriteEntityIds.toList()` and checks if it is part of `settingsWearViewModel.favoriteEntityIds.toList()`. 

So what is the goal of `isFavoriteSelected(index)`? Is it to verify if the item is still in the list? Shouldn't it always return true for any Row created from the `favoritesList`? At least I'm not really following what is going on, or why this check is happening haha

Reply:
ok i see the issue now where it duplicates....its due to the `Row` itself and actually it should always return `false` because its already selected as its part of favorites. I was able to reproduce and correct this. The `isFavoriteSelected` is actually important to keep track of the checked state itself but it was causing a weird loop in that one area. Good catch!!